src/hotspot/share/opto/loopopts.cpp

```diff
@@ -228,6 +228,40 @@ Node* PhaseIdealLoop::split_thru_phi(Node* n, Node* region, int policy) {
     }
   }
 
+  // If the region is a Loop, we are removing the old n,
+  // and need to yank it from the _body. If any phi we
+  // just split through now has no use any more, it also
+  // has to be removed.
+  IdealLoopTree* region_loop = get_loop(region);
```

Comment:
The method is already quite large. This could probably nicely be extracted to a "yank_old_nodes()" method or something like that. You can keep the ` _igvn.replace_node(n, phi)` in this method.